system/Entity/Entity.php

```diff
@@ -439,12 +439,21 @@ public function __set(string $key, $value = null)
 
         $value = $this->castAs($value, $key, 'set');
 
-        // if a set* method exists for this key, use that method to
+        // if a setter method exists for this key, use that method to
         // insert this value. should be outside $isNullable check,
         // so maybe wants to do sth with null value automatically
         $method = 'set' . str_replace(' ', '', ucwords(str_replace(['-', '_'], ' ', $key)));
 
-        if (method_exists($this, $method)) {
+        // If a "`_set` + $key" method exists, it is a setter.
+        if (method_exists($this, '_' . $method)) {
+            $this->{$method}($value);
+
+            return $this;
+        }
+
+        // If a "`set` + $key" method exists, it is also a setter.
+        // But setAttributes() is not a setter for this.
+        if ($key !== 'attributes' && method_exists($this, $method)) {
```

Comment:
```suggestion
        // If a "`_set` + $key" method exists, it is a setter,  or
        // if a "`set` + $key" method exists, it is also a setter.
        // But setAttributes() is not a setter for this.
        if (method_exists($this, '_' . $method) || ($key !== 'attributes' && method_exists($this, $method))) {
```

Reply:
Let's make code shorter,

Reply:
Thanks, I found a bug in my code.

Why do you like shorter code?